lib/ansible/plugins/action/__init__.py

```diff
@@ -86,6 +89,17 @@ def run(self, tmp=None, task_vars=None):
                                  ' the tmpdir']
         del tmp
 
+        # set info needed to make decisions, fallback for old custom plugins w/o get_option
+        try:
+            self._admin_users = self._connection._shell.get_option('admin_users')
+        except AnsibleError:
+            self._admin_users = ['root']
+        try:
+            remote_user = self._connection.get_option('remote_user')
+        except AnsibleError:
+            remote_user = self._play_context.remote_user
+        self._deescalate = bool(self._play_context.become and self._play_context.become_user not in self._admin_users + [remote_user])
```

Comment:
Setting these things here still feels wrong.  can you walk me through where something isn't being "finalized" when __init__ is run so that I can figure out why this is needed and if there's some other way that might be better?

Reply:
the better way involves some rewrites in task_executor, which i'm planning, but they are out of scope for this PR